internal/adapters/events/events_forwarder.go

```diff
@@ -217,12 +220,17 @@ func fromPlayerEventTypeToGrpcPlayerEventType(eventType events.PlayerEventType)
 	}
 }
 
-func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) error {
+// handlerGrpcClientResponse checks if the gRPC response had unexpected communication or network errors.
+func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) (int32, error) {
 	if err != nil {
-		return err
-	}
-	if eventResponse.Code != 200 {
-		return errors.NewErrUnexpected("failed to forward event room at \"%s\"", forwarder.Name)
+		grpcStatus, ok := status.FromError(err)
+		if !ok {
+			return eventResponse.Code, errors.NewErrUnexpected("failed to forward event room at \"%s\" with unknown grpc code: %w", forwarder.Name, err)
```

Comment:
why don't we return `grpcStatus` and let go of this custom `.Code`? The function is already specific to grpc

Reply:
In this specific case, I opted for the `.Code` due to the parse to status failing

I'm using the .Code whenever the status.Code isn't available